lib/rubygems/defaults.rb

```diff
@@ -147,13 +147,6 @@ def self.default_cert_path
     File.join Gem.user_home, ".gem", "gem-public_cert.pem"
   end
 
-  ##
-  # Whether to expect full paths in default gems - true for non-MRI
-  # ruby implementations
-  def self.default_gems_use_full_paths?
-    ruby_engine != 'ruby'
-  end
```

Comment:
I also wanted to discuss this particular removal. The more conservate approach would've been that add an exception for  `jruby` here, since I'm not sure whether other implementations actually rely on this.